### PR TITLE
Add user login

### PR DESCRIPTION
**Pull request by ahominal**
- add user login (functional register and screen)
- add password recovery option for already registered user
- add some to both new functionalities

### CODE REVIEW

File: app/src/androidTest/java/com/sdp/swiftwallet/UiTest/ForgotPwActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    Intents.init();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    Intents.release();
+  }
+
+  @Test
+  public void test_display(){
```
Comment (0xdieman) [inline app/src/androidTest/java/com/sdp/swiftwallet/UiTest/ForgotPwActivityTest.java:40]: I talked about that in Jerome's PR and we should discuss it but I think `mixedCase` is the convention for tests naming

**Review by 0xdieman — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/swiftwallet/UiTest/ForgotPwActivityTest.java (hunk 1 above)
Comment (ahominal) [reply app/src/androidTest/java/com/sdp/swiftwallet/UiTest/ForgotPwActivityTest.java:40]: thanks for clarification i'll go and check

File: app/src/main/java/com/sdp/swiftwallet/data/repository/FirebaseAuthImpl.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+   * Creates a User in the Firebase database
+   * @param username his username
+   * @param email his email
+   * @param password his password
+   * @param registerActivity from register activity
+   * @param resultActivity to login activity back
+   */
+    public void createUserWithEmailAndPassword(String username, String email, String password, Activity registerActivity, Activity resultActivity) {
+        firebaseAuth.createUserWithEmailAndPassword(email, password)
+                .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
```
[the remark this answers is not in the document]
Comment (ahominal) [reply app/src/main/java/com/sdp/swiftwallet/data/repository/FirebaseAuthImpl.java:55]: Yes sure! good suggestion, i put a refactor task on my next sprint

**Review by vfaltings — APPROVED** [review]
Looks good to me

**Pull request merged**